File: src/color.c
/*
 * XENLY - high-level and general-purpose programming language
 * created, designed, and developed by Cyril John Magayaga (cjmagayaga957@gmail.com, cyrilmagayaga@proton.me).
 *
 * It is initially written in C programming language.
 *
 */
#include <stdio.h>
#include "color.h"

// Red text
void red() {
  printf("\033[1;31m");
}

// Green text
void green() {
  printf("\033[1;32m");
}

// Yellow text
void yellow() {
  printf("\033[1;33m");
}

// Blue text
void blue() {
  printf("\033[1;34m");
}

// Purple text
void purple() {
  printf("\033[1;35m");
}

// White text
void white() {
  printf("\033[1;37m");
}

// Orange text
void orange() {
  printf("\033[1;38;5;208m");
}

// Cyan text
void cyan() {
  printf("\033[1;36m");
}

// Reset color
void resetColor() {
  printf("\033[0m");
}

// Red background
void setBackgroundRed() {
  printf("\033[41m");
}

// Green background
void setBackgroundGreen() {
  printf("\033[42m");
}

// Yellow background
void setBackgroundYellow() {
  printf("\033[43m");
}

// Blue background
void setBackgroundBlue() {
  printf("\033[44m");
}

// Purple background
void setBackgroundPurple() {
  printf("\033[45m");
}

// White background
void setBackgroundWhite() {
  printf("\033[47m");
}

// Reset background color
void resetBackgroundColor() {
  printf("\033[49m");
}

// Black text and Orange background
void blackAndOrange() {
  printf("\033[0;30;48;5;208m");
}

File: src/color.rs
pub fn red() {
    print!("\x1B[1;31m");
}

pub fn green() {
    print!("\x1B[1;32m");
}

pub fn yellow() {
    print!("\x1B[1;33m");
}

pub fn blue() {
    print!("\x1B[1;34m");
}

pub fn purple() {
    print!("\x1B[1;35m");
}

pub fn white() {
    print!("\x1B[1;37m");
}

pub fn orange() {
    print!("\x1B[1;38;5;208m");
}

pub fn cyan() {
    print!("\x1B[1;36m");
}

pub fn reset_color() {
    print!("\x1B[0m");
}

pub fn set_background_red() {
    print!("\x1B[41m");
}

pub fn set_background_green() {
    print!("\x1B[42m");
}

pub fn set_background_yellow() {
    print!("\x1B[43m");
}

pub fn set_background_blue() {
    print!("\x1B[44m");
}

pub fn set_background_purple() {
    print!("\x1B[45m");
}

pub fn set_background_white() {
    print!("\x1B[47m");
}

pub fn reset_background_color() {
    print!("\x1B[49m");
}

pub fn black_and_orange() {
    print!("\x1B[0;30;48;5;208m");
}
